src/main/java/com/softserve/academy/spaced/repetition/controller/CardController.java

```diff
@@ -133,41 +139,45 @@
                                                                    @PathVariable Long deckId,
                                                                    @PathVariable Long cardId,
-                                                                   @Validated(Request.class) @RequestBody Card card) {
-        cardService.updateCard(cardId, card);
-        Link selfLink = linkTo(methodOn(CardController.class)
-                .getCardByCourseAndDeck(categoryId, courseId, deckId, card.getId())).withSelfRel();
+                                                                   @Validated(Request.class) @RequestBody Card card,
+                                                                   @RequestParam(required = false) List<String> images) {
+
+        cardService.updateCard(card, deckId, images);
+        Link selfLink = linkTo(methodOn(CardController.class).getCardByCourseAndDeck(categoryId, courseId, deckId, card.getId())).withSelfRel();
+
         CardPublicDTO cardPublicDTO = DTOBuilder.buildDtoForEntity(card, CardPublicDTO.class, selfLink);
         return new ResponseEntity<>(cardPublicDTO, HttpStatus.OK);
     }
 
     @Auditable(action = AuditingAction.EDIT_CARD_VIA_CATEGORY_AND_DECK)
     @PutMapping(value = "/api/category/{categoryId}/decks/{deckId}/cards/{cardId}")
     @PreAuthorize(value = "@accessToUrlService.hasAccessToCard(#categoryId, #deckId, #cardId)")
-    public ResponseEntity<CardPublicDTO> updateCardByCategoryAndDeck(@PathVariable Long categoryId,
-                                                                     @PathVariable Long deckId,
-                                                                     @PathVariable Long cardId,
-                                                                     @Validated(Request.class) @RequestBody Card card) {
-        cardService.updateCard(cardId, card);
-        Link selfLink = linkTo(methodOn(CardController.class)
-                .getCardByCategoryAndDeck(categoryId, deckId, card.getId())).withSelfRel();
+    public ResponseEntity<CardPublicDTO> updateCardByCategoryAndDeck(@PathVariable Long categoryId, @PathVariable Long deckId, @PathVariable Long cardId, @Validated(Request.class) @RequestBody Card card, @RequestParam(required = false) List<String> images) {
+        cardService.updateCard(card, deckId, images);
+        Link selfLink = linkTo(methodOn(CardController.class).getCardByCategoryAndDeck(categoryId, deckId, card.getId())).withSelfRel();
+
         CardPublicDTO cardPublicDTO = DTOBuilder.buildDtoForEntity(card, CardPublicDTO.class, selfLink);
         return new ResponseEntity<>(cardPublicDTO, HttpStatus.OK);
     }
 
     @Auditable(action = AuditingAction.EDIT_CARD_VIA_CATEGORY_AND_DECK)
-    @PutMapping(value = "/api/decks/{deckId}/cards/{cardId}")
+    @PostMapping(value = "/api/decks/{deckId}/cards/{cardId}")
     @PreAuthorize(value = "@accessToUrlService.hasAccessToCard(#deckId, #cardId)")
```

Comment:
You must use PUT Mapping by REST API convension

Reply:
in another task